### ISSUE TITLE
Dotenv files are respected and incorrectly override environment variables

### ISSUE THREAD

**Issue by frederikhors** [ory/kratos#1856]
**Describe the bug**

If I have ENVironment variables and a `.env` file in root where kratos starts it reads the .env file variable instead of the environment variable.

**Reproducing the bug**

1. Run `export DSN="varFromEnv"`
1. Create `.env` file with: `DSN="varFromFile"`
1. Run `./kratos -c ./kratos.yml migrate sql -e --yes`

The value of DSN in Kratos is: `varFromFile`.

**Expected behavior**

I think we should read **FIRST** from Environment variable (if present) because `.env` file can contain development vars (as in my case).

**Environment**

- Version: oryd/kratos:v0.7.6-alpha.1
- Environment: Docker

**Comment by aeneasr** [comment]
Wouldn't that be the same as setting `dsn: ...` in `kratos.yml`?

**Comment by frederikhors** [comment]
> I don't think we'll support `.env` files in the future

You are already doing it!

I opened this issue for this reason.

I'm just asking you to let the environment variables prevail over everything else.

**Comment by aeneasr** [comment]
> You are already doing it!

That is not true. We do not support `.env` files in this project. And I don't know what "gotrue" is, we are not using this library.

**Comment by aeneasr** [comment]
> Ok, but at least the environment variables have priority over those of the config?

Yes! :)

> And using this, does foo come before the bar or the last one wins?

Ha, good question! I don't know to be honest. My guess would be last one wins!

**Comment by frederikhors** [comment]
> > Ok, but at least the environment variables have priority over those of the config?
> 
> Yes! :)

@aeneasr I think they don't. At least `LOG_LEVEL` one.

Using this config:

```yml
log:
  format: text
  level: debug
```

and this env:

```
LOG_LEVEL=error
```

makes the log print DEBUG info.

I expect `error` to win here.

**Comment by kmherrmann** [comment]
Closed due to inactivity.

**Comment by frederikhors** [comment]
@kmherrmann what does it mean "inactivity"? This is still an issue... Can you please re-open?

**Comment by kmherrmann** [comment]
OK - sorry about that, am trying to clean up old issues, and this one last saw activity in 2021.
Thanks for the feedback, reopening!

**Comment by frederikhors** [comment]
Ok. I will monitor and close this when is fixed. Don't worry. I'm active.
